aws_lambda_powertools/utilities/data_classes/cloud_watch_alarm_event.py

```diff
@@ -0,0 +1,206 @@
+    def expression(self) -> Optional[str]:
+        """
+        The mathematical expression for calculating the metric, if applicable.
+        """
+        return self.get("expression", None)
+
+    @property
+    def label(self) -> Optional[str]:
+        """
+        Optional label of the metric.
+        """
+        return self.get("label", None)
+
+    @property
+    def namespace(self) -> Optional[str]:
+        """
+        Namespace of the correspondent CloudWatch Metric.
+        """
+        if self._metric_stat is not None:
+            return self._metric_stat.get("metric", {}).get("namespace", None)
+
+        return None
+
+    @property
+    def name(self) -> Optional[str]:
+        """
+        Name of the correspondent CloudWatch Metric.
+        """
+        if self._metric_stat is not None:
+            return self._metric_stat.get("metric", {}).get("name", None)
+
+        return None
+
+    @property
+    def dimensions(self) -> Optional[dict]:
+        """
+        Additional dimensions of the correspondent CloudWatch Metric, if available.
+        """
+        if self._metric_stat is not None:
+            return self._metric_stat.get("metric", {}).get("dimensions", None)
+
+        return None
+
+    @property
+    def period(self) -> Optional[int]:
+        """
+        Metric evaluation period, in seconds.
+        """
+        if self._metric_stat is not None:
+            return self._metric_stat.get("period", None)
+
+        return None
+
+    @property
+    def stat(self) -> Optional[str]:
+        """
+        Statistical aggregation of metric points, e.g. Average, SampleCount, etc.
+        """
+        if self._metric_stat is not None:
+            return self._metric_stat.get("stat", None)
+
+        return None
+
+    @property
+    def return_data(self) -> bool:
+        """
+        Whether this metric data is used to determine the state of the alarm or not.
+        """
+        return self["returnData"]
+
+
+class CloudWatchAlarmEvent(DictWrapper):
+    @property
+    def source(self) -> str:
+        """
+        Source of the triggered event, usually it is "aws.cloudwatch".
+        """
+        return self["source"]
+
+    @property
+    def alarm_arn(self) -> str:
+        """
+        The ARN of the CloudWatch Alarm.
+        """
+        return self["alarmArn"]
+
+    @property
+    def region(self) -> str:
+        """
+        The AWS region in which the Alarm is active.
+        """
+        return self["region"]
+
+    @property
+    def source_account_id(self) -> str:
+        """
+        The AWS Account ID that the Alarm is deployed to.
+        """
+        return self["accountId"]
+
+    @property
+    def timestamp(self) -> str:
+        """
+        Alarm state change event timestamp in ISO-8601 format.
+        """
+        return self["time"]
+
+    @property
+    def alarm_name(self) -> str:
+        """
+        Alarm name.
+        """
+        return self["alarmData"]["alarmName"]
```

Comment:
I think we need to create a new `AlarmData` class and then customers will be able to access `event.alarm_data.alarm_name`. It is valid for all fields nested in the `alarm_data` key. E.g `event.alarm_data.state.value`.